adaptdl/adaptdl/speedup.py

```diff
@@ -232,28 +298,32 @@ def fit(nodes, replicas, local_bsz, step_time, step_time_compute):
         params[3] = upper[3] = lower[3]
         params[5] = upper[5] = lower[5]
     bounds = scipy.optimize.Bounds(lower, upper, keep_feasible=True)
-
-    args = (nodes, replicas, local_bsz, step_time, step_time_compute)
+    args = (nodes, replicas, local_bsz, accumulation_steps,
+            step_time, step_time_compute, accumulation_time)
     # FIXME: need to handle optimization failures and propagate to the Trainer.
     grad_fn = autograd.grad(_obj_fn)
     result = scipy.optimize.minimize(_obj_fn, params, args=args,
                                      jac=grad_fn, bounds=bounds)
     params = result.x
-
     np = orig_np  # Restore original numpy.
     return Params(*params)
 
 
-def _predict_log(params, nodes, replicas, local_bsz):
+def _predict_log(params, nodes, replicas, local_bsz, accumulation_steps):
     params = Params(*params)
     step_time_compute = _predict_compute(params, local_bsz)
     step_time_network = _predict_network(params, nodes, replicas)
+    step_time_accumulation = step_time_compute
     gamma = params.gamma
     # Return predicted total step time in log-space to avoid numerical issues
     # in autograd and optimization.
-    return (np.log(step_time_compute ** gamma +
-                   step_time_network ** gamma) / gamma,
-            step_time_compute, step_time_network)
+    step_time = np.log(
+        ((step_time_compute) ** gamma + step_time_network ** gamma) **
+        (1 / gamma) + step_time_accumulation * accumulation_steps)
```

Comment:
I'm not sure about the reason to take the log now that we're exponentiating by (1/gamma) anyways, maybe fine now to just return the prediction in linear space.

Reply:
Changed